File: Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/data_preprocessing_template.R
# Data Preprocessing Template

# Importing the dataset
dataset = read.csv('Data.csv')
### JUST IN NEED WE WILL NEED TO TAKE A SUBSET OF THE IMPORTED DATASET.
# dataset = [, 2:3]

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

###  IN THE MOST OF THE TIMES WE WILL NOT NEED TO DO THE FEATURE SCALING,
### BECAUSE IT WILL BE A PART OF THE ALGORITHM.
# # Feature Scaling
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

File: Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/data_preprocessing_template.py
# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, -1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

""" IN THE MOST OF THE TIMES WE WILL NOT NEED TO DO THE FEATURE SCALING, BECAUSE IT WILL BE A PART OF THE ALGORITHM.
# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)

# We do not need to do the feature scaling on the dependent variable, since it is categorical.
# But if we had to, that is how it would be done:
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)
y_test = sc_y.transform(y_test)
"""